plugins/modules/architectures.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import requests
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: architectures
+
+short_description: Query OpenShift architectures
+
+version_added: "1.0.0"
+
+description: Query supported architectures for a given OpenShift version
+
+options:
+  openshift_version:
+    description: Version of OpenShift
+    default: None
+    required: True
+    type: string
+
+author:
+    - Chris Wheeler (@clwheel)
+"""
+
+EXAMPLES = r"""
+- name: Query OpenShift architectures
+  architectures:
+    openshift_version: 4.16.19
+"""
+
+RETURN = r"""
+architectures:
+  description: A list of supported OpenShift architectures
+  type: dict
+  returned: always
+  sample: { "ARM64_ARCHITECTURE": "supported",
+            "MULTIARCH_RELEASE_IMAGE": "tech-preview",
+            "PPC64LE_ARCHITECTURE": "supported",
+            "S390X_ARCHITECTURE": "supported",
+            "X86_64_ARCHITECTURE": "supported" }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+QUERY_PARAMS_LIST = ["openshift_version"]
+
+def run_module():
+    module_args = dict(
+        openshift_version = dict(type="str", required=True),
+    )
+
+    token = os.environ.get('AI_API_TOKEN')
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
+
+    # Set headers
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+
+    query_params = {}
+
+    for k in QUERY_PARAMS_LIST:
+        val = module.params.get(k)
+        if val:
+            query_params = query_params | { k: val }
+
+    response = requests.get(f"{API_URL}/support-levels/architectures", params=query_params, headers=headers)
```

Comment:
The query_param could be avoided since is only one param that is supported.

```suggestion
    response = requests.get(f"{API_URL}/support-levels/architectures", params={"openshift_version": module.params.get('openshift_version')}, headers=headers)
```

Reply:
True, but I would prefer to keep it consistent with how other modules are implemented and extensible in case additional query options are added to the API in the future.